sdks/python/apache_beam/io/filesystem.py

```diff
@@ -576,6 +576,27 @@ def exists(self, path):
     """
     raise NotImplementedError
 
+  @abc.abstractmethod
+  def checksum(self, path):
+    """Fetch checksum metadata of a file on the FileSystem.
+
+    This operation returns checksum metadata as stored in the underlying
+    FileSystem. It should not need to read file data to obtain this value.
```

Comment:
<!--new_thread; commit:24c81c1018b47c5316f09f3b1eef8725b5492848; resolved:0-->
``FileSystem`` (with pydoc tags) (here and below)